crypto/keys/private_keys.go

```diff
@@ -168,3 +169,15 @@ func MarshalPrivateKey(key crypto.Signer) ([]byte, error) {
 
 	return nil, fmt.Errorf("unsupported key type: %T", key)
 }
+
+// NewFromPKCS11Config returns a crypto.Signer that uses a PKCS#11 interface.
+func NewFromPKCS11Config(modulePath string, config *keyspb.PKCS11Config) (crypto.Signer, error) {
+	if modulePath == "" {
+		return nil, errors.New("No PKCS#11 module path set, cannot create signer")
+	}
+	pubKey, err := NewFromPublicPEM(config.GetPublicKey())
+	if err != nil {
+		return nil, fmt.Errorf("Failed to load public key from %q: %s", config.GetPublicKey(), err)
```

Comment:
Error strings should begin with a lowercase letter: https://github.com/golang/go/wiki/CodeReviewComments#error-strings